### PR TITLE
[NFC][mlir][scf] Fix SCF dialect's operations' descriptions

### PR DESCRIPTION
**Pull request by EtoAndruwa**
- Added the dialect's prefix to operations' descriptions to follow the same style inside the TableGen file.
- Minor changes in some operations' descriptions.

### CODE REVIEW

File: mlir/include/mlir/Dialect/SCF/IR/SCFOps.td [hunk 1]
```diff
@@ -875,25 +875,25 @@ def ReduceOp : SCF_Op<"reduce", [
     DeclareOpInterfaceMethods<RegionBranchTerminatorOpInterface>]> {
   let summary = "reduce operation for scf.parallel";
   let description = [{
-    "scf.reduce" is the terminator for "scf.parallel" operations. It can model
+    The `scf.reduce` is the terminator for `scf.parallel` operations. It can model
```
Comment (matthias-springer) [inline mlir/include/mlir/Dialect/SCF/IR/SCFOps.td:878]: Nit: I think there should be no `The` here. Or add `operation`.

File: mlir/include/mlir/Dialect/SCF/IR/SCFOps.td [hunk 2]
```diff
@@ -929,9 +929,9 @@ def ReduceReturnOp :
     SCF_Op<"reduce.return", [HasParent<"ReduceOp">, Pure, Terminator]> {
   let summary = "terminator for reduce operation";
   let description = [{
-    "scf.reduce.return" is a special terminator operation for the block inside
-    "scf.reduce" regions. It terminates the region. It should have the same
-    operand type as the corresponding operand of the enclosing "scf.reduce" op.
+    The `scf.reduce.return` is a special terminator operation for the block inside
```
Comment (matthias-springer) [inline mlir/include/mlir/Dialect/SCF/IR/SCFOps.td:932]: No `The` or add `operation`.

File: mlir/include/mlir/Dialect/SCF/IR/SCFOps.td [hunk 3]
```diff
@@ -1172,12 +1172,12 @@ def YieldOp : SCF_Op<"yield", [Pure, ReturnLike, Terminator,
                  "WhileOp"]>]> {
   let summary = "loop yield and termination operation";
   let description = [{
-    "scf.yield" yields an SSA value from the SCF dialect op region and
+    The `scf.yield` yields an SSA value from the SCF dialect op region and
```
Comment (matthias-springer) [inline mlir/include/mlir/Dialect/SCF/IR/SCFOps.td:1175]: No `The` or add `operation`.

**Review by matthias-springer — APPROVED** [review]

File: mlir/include/mlir/Dialect/SCF/IR/SCFOps.td (hunk 1 above)
Comment (EtoAndruwa) [reply mlir/include/mlir/Dialect/SCF/IR/SCFOps.td]: Hello! Thank you, missed `operation`. Fixed.

**Comment by EtoAndruwa** [comment]
Hello, @matthias-springer! Can you please merge this?

**Pull request merged**